src/server/pmix_server.c

```diff
@@ -688,6 +688,7 @@ PMIX_EXPORT pmix_status_t PMIx_server_init(pmix_server_module_t *module,
             return rc;
         }
         /* restore our original primary server */
+        PMIX_LOAD_PROCID(&myproc, pmix_globals.myid.nspace, pmix_globals.myid.rank);
         rc = PMIx_tool_set_server(&myproc, NULL, 0);
```

Comment:
Why don't we just remove the `myproc` variable and pass `&pmix_globals.myid`? No point in having that variable just to pass here.

Reply:
@drwootton just pushed this fix. I think we are good to go now.